js/lex.go

```diff
@@ -33,6 +33,30 @@ const (
 	TemplateToken
 )
 
+// TokenState determines a state in which next token should be read
+type TokenState uint32
+
+// TokenState values
+const (
+	ExprState TokenState = iota
+	StmtParensState
+	SubscriptState
+	PropNameState
+)
```

Comment:
Just made up. What do you find unclear about them? It says "expression state", "statement parentheses state", "subscript state", "property name state" to be as explicit about where it is as possible :)